### ISSUE TITLE
Markdown support for creating new posts

### ISSUE THREAD

**Issue by kuldeepjambhulkar** [krishnaacharyaa/wanderlust#57]
### Description

As of now, creating post is just typing plain text in the description box.
This doesn't provide any control over text formatting, like making text bold, italic, or HTML formatting like heading, etc.
It would be a better experience for the writer's as well to be in the control of these things.

### Use Case

As a blog writer
I want to write the content in markdown format, instead of plain text
So that, it'll provide more control in the hands of the writer on how the text should be displayed on UI

**Comment by chinmaykunkikar** [comment]
That can significantly improve the overall UX. What do you think, @krishnaacharyaa?

Also, this can be a great way to introduce and work with [MDX](https://mdxjs.com/) :)

**Comment by krishnaacharyaa** [comment]
Yeah sure @chinmaykunkikar, this is great! But I personally feel we can focus on the user auth first as it is quite a big feature, so that we can add features like edit and delete posts and later we can work on this, But if we are looking it as, we will complete the smaller tasks which will enhance the UX first we can work on this, I'm open for this even. Kindly let me know your thoughts.

**Comment by chinmaykunkikar** [comment]
I agree. Auth will fundamentally change things. That will enhance other features and make them feel more personalized. Let's focus on auth for the time being.

**Comment by krishnaacharyaa** [comment]
Sure, I have set the milestone for the same. I have the opinion that we can focus on the test cases first, As it will set a baseline for the code quality and we can set code coverage, And then work on the features like user auth, by this we make sure that every new feature comes with the test cases along with it, so that we are not in feat of breaking the existing code when we add a new feature. The feature may be small, but I feel this is the right approach moving forward, let me know your thoughts :)

**Comment by AnirudhMemani** [comment]
I can do this

**Comment by krishnaacharyaa** [comment]
Sure @AnirudhMemani 
Please let us know what you are gonna do, probably a image of the image of it 
And ETA

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Probably by Monday
![Screenshot 2024-04-20 001028](https://github.com/krishnaacharyaa/wanderlust/assets/46074384/7a643420-d693-4950-bfa1-bbeed0b82804)

**Comment by krishnaacharyaa** [comment]
Okay @AnirudhMemani please take it
